apps/emqx_utils/src/emqx_utils_maps.erl

```diff
@@ -293,3 +294,12 @@ if_only_to_toggle_enable(OldConf, Conf) ->
         {_, _, _} ->
             false
     end.
+
+%% Like `maps:update_with', but does nothing if key does not exist.
+update_if_present(Key, Fun, Map) ->
+    case Map of
+        #{Key := Val} ->
+            Map#{Key := Fun(Val)};
+        _ ->
+            Map
+    end.
```

Comment:
```
update_if_present(Key, Fun, Map) when is_map_key(Key, Map) ->
    Map#{Key := Fun(Val)};
update_if_present(_Key, _Fun, Map) ->
    Map.
```

Reply:
We still need to extract the value from the map, so I wonder if it's better with clauses? :thinking: